src/parser.c

```diff
@@ -203,10 +203,8 @@ int ecs_parse_component_expr(
 
         if (prev_is_0) {
             /* 0 can only apppear by itself */
-            char err[50] = "0 can only apppear by itself";
-            ecs_print_error_string(err, sig, system_id, bptr);
-
-            ecs_abort(ECS_INVALID_SIGNATURE, err);
+            ecs_print_error_string(ecs_strerror(ECS_ZERO_CAN_ONLY_APPEAR_BY_ITSELF), sig, system_id, bptr);
```

Comment:
I think (but I can be persuaded if there are good reasons not to) that I would prefer a printf-style formatted string here, so that you could do something like (example):

```
ecs_print_error_string(sig, system_id, bptr, "unresolved component name '%s'", component_name);
```
In the current setup you print the same error twice (in the `ecs_print_error_string` and in the `ecs_abort`) and the error message is constrained by what limited parameters you can pass into either.

Reply:
I need  to turn `ecs_print_error_string` into a variadic macro. 

I'll try to send a PR tonight.

Reply:
Done, please take a look. Now the message is 
```
Move at argument #2. Error: "unresolved component name 'Velocity'"
Position, Velocity
~~~~~~~~~~^
[err] abort /Users/pabloweremczuk/Documents/Proyectos/cpp/flecs/src/parser.c:276: invalid component id (Velocity)
```

Reply:
That looks awesome! taking a look

Reply:
Please let me know if more changes are required. My plan is to work on the tests this weekend.

Thanks.